### ISSUE TITLE
DataTable: restoreState should check if pagination related options exist in storage

### ISSUE THREAD

**Issue by MBelniak** [primefaces/primereact#6424]
### Describe the bug

See https://github.com/primefaces/primereact/blob/master/components/lib/datatable/DataTable.js#L292
There is no check if 'first' and 'rows' exist in the object taken from storage. In my case, I do not display paginator when there is no data, thus the storage does not contain keys related to pagination. Then, the data may arrive at some point, thus I display the pagination and DataTable incorrectly assumes, that 'first' and 'rows' are there in the storage which results in NaN being put as current page and pageCount, which completely breaks the table.

### PrimeReact version

10.3.3

### React version

18.x

### Language

TypeScript

### Build / Runtime

Create React App (CRA)

**Comment by Rekl0w** [comment]
Can you provide a link for this from stackblitz ?

**Comment by MBelniak** [comment]
After a deep dive into the code I realized that one needs to provide 'first' and 'rows' props to the DataTable when onPage is provided. In my case I provided onPage, but I didn't provide first and rows.
Also, I see now that I shouldn't use local storage and at the same time display paginator conditionally, because it will erase my pagination state each time I refresh the page (unless I provide custom saveState and restoreState and save first and rows even if pagination is not displayed). 

Stackblitz for reference: https://stackblitz.com/edit/vitejs-vite-lrfnmv?file=src%2FApp.tsx,package-lock.json,node_modules%2Fprimereact%2Fdatatable%2Fdatatable.d.ts,node_modules%2Fprimereact%2Fpaginator%2Fpaginator.d.ts,node_modules%2F%40types%2Freact%2Fts5.0%2Findex.d.ts&terminal=dev

I'm closing this one as there is nothing to be fixed on primereact's side.
